src/engine/enginemaster.cpp

```diff
@@ -443,48 +488,184 @@ void EngineMaster::process(const int iBufferSize) {
+            // to be able to process the same effects on different buffers
+            // within the same callback. For consistency between the MicMonitorModes,
+            // process master effects here before mixing in talkover.
+            applyMasterEffects(iBufferSize, iSampleRate);
+
+            // Mix talkover with master
+            if (m_pNumMicsConfigured->get() > 0) {
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
+                    m_pTalkover, 1.0,
                     iBufferSize);
-        }
+            }
 
-        // Process master channel effects
-        if (m_pEngineEffectsManager) {
-            GroupFeatureState masterFeatures;
-            // Well, this is delayed by one buffer (it's dependent on the
-            // output). Oh well.
-            if (m_pVumeter != NULL) {
-                m_pVumeter->collectFeatures(&masterFeatures);
+            // Copy master mix (with talkover mixed in) to booth output with booth gain
+            if (boothEnabled) {
+                CSAMPLE boothGain = m_pBoothGain->get();
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+                m_boothGainOld = boothGain;
+            }
+
+            // Apply master gain
+            // TODO(Be): make this not affect the headphones. Refer to
+            // https://bugs.launchpad.net/mixxx/+bug/1458213
+            CSAMPLE master_gain = m_pMasterGain->get();
+            if (m_bRampingGain) {
+                SampleUtil::applyRampingGain(m_pMaster, m_masterGainOld, master_gain,
+                                             iBufferSize);
+            } else {
+                SampleUtil::applyGain(m_pMaster, master_gain, iBufferSize);
+            }
+            m_masterGainOld = master_gain;
+
+            // Record/broadcast signal is the same as the master output
+            m_ppSidechainOutput = &m_pMaster;
+        } else if (configuredMicMonitorMode == MicMonitorMode::DIRECT_MONITOR) {
+            // Skip mixing talkover with the master and booth outputs
+            // if using direct monitoring because it is being mixed in hardware
+            // without the latency of sending the signal into Mixxx for processing.
+            // However, include the talkover mix in the record/broadcast signal.
+
+            // Copy master mix to booth output with booth gain
+            if (boothEnabled) {
+                CSAMPLE boothGain = m_pBoothGain->get();
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
```

Comment:
I do not think we can compare us 1:1 to hardware Mixers. They should have a very low round-trip latency. So the do not have the issue of an annoying echo in the headphones. They have the option to remove the microphones from the booth mainly to avoid a feedback loop. 
This cannot happen using Headphones, so there is no strong need to remove the sound from there. 
For me it is sufficient to just pass booth to the Headphone Mixxx.
If you think we need an option for this we can add this as an option to Mic Monitor Mode 
* Master output only
* Master and headphone (master cue)
* Master, booth and headphone (master cue)
* Direct monitor (recording and broadcasting only)